src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -165,11 +165,24 @@ public String createDataset(
     String description = "Create dataset " + datasetRequest.getName();
     UUID defaultProfileId = datasetRequest.getDefaultProfileId();
     loggingMetrics.set(BardEventProperties.BILLING_PROFILE_ID_FIELD_NAME, defaultProfileId);
+
+    // Locate billing profile in TDR or Rawls
+    // No auth check: Just a check if there is an entry in our db for this billing profile
+    boolean isTdrBillingProfile = false;
+    try {
+      profileService.getProfileByIdNoCheck(defaultProfileId);
+      isTdrBillingProfile = true;
+    } catch (ProfileNotFoundException ex) {
+      // Assume that the billing project lives in Rawls
+      // AuthorizeRawlsBillingProjectsUseStep will return the relevant error message if not
+    }
```

Comment:
If you wanted to, you could remove the empty comment by moving `false` into the `else`
```suggestion
    final boolean isTdrBillingProfile;
    try {
      profileService.getProfileByIdNoCheck(defaultProfileId);
      isTdrBillingProfile = true;
    } catch (ProfileNotFoundException ex) {
      isTdrBillingProfile = false;
    }
```